src/lib/DutchDecayLib.sol

```diff
@@ -68,6 +68,35 @@ library DutchDecayLib {
         }
     }
 
+    /// @notice returns the linear interpolation between the two points
+    /// @param startPoint The start of the decay
+    /// @param endPoint The end of the decay
+    /// @param currentPoint The current position in the decay
+    /// @param startAmount The amount of the start of the decay
+    /// @param endAmount The amount of the end of the decay
+    function linearDecay(
+        uint256 startPoint,
+        uint256 endPoint,
+        uint256 currentPoint,
+        int256 startAmount,
+        int256 endAmount
+    ) internal pure returns (int256) {
+        if (currentPoint >= endPoint) {
+            return endAmount;
+        }
+        unchecked {
+            uint256 elapsed = currentPoint - startPoint;
```

Comment:
do we need to check for startPoint > currentPoint to avoid overflow here?

Reply:
Good question. It's not possible with the current code since the caller passes in either 0 or `relativeBlocks.getElement(prev)` for the `startPoint` where `relativeBlocks.getElement(prev)` is guaranteed to be less than the `currentPoint`.
Similarly it's not possible for the `startPoint` to be greater than or equal to the `endPoint`. It could be equal if (currentPoint > endPoint) but we check that above. Otherwise, the `locateCurvePosition()` call ensures that it halts the search as soon as it finds a block that is greater than or equal to the `blockDelta`, meaning every value that came before it would be less than the `blockDelta` (and transitively less than the block it finds).

Reply:
It IS possible for this int256 cast to overflow if `startAmount` is max int256 and `endAmount` is negative:
`int256(0 - uint256(startAmount - endAmount).mulDivDown(elapsed, duration)`

Going to remove the `unchecked`.